src/Checkout/ExpressCheckout/SalesChannel/ExpressCreateOrderRoute.php

```diff
@@ -69,8 +73,26 @@ public function createPayPalOrder(Request $request, SalesChannelContext $salesCh
 
             $this->logger->debug('Building order');
             $order = $this->paypalOrderBuilder->getOrderFromCart($cart, $salesChannelContext, new RequestDataBag($request->request->all()));
-            $order->getPaymentSource()?->getPaypal()?->getExperienceContext()->setShippingPreference(ApplicationContext::SHIPPING_PREFERENCE_GET_FROM_FILE);
-            $order->getPaymentSource()?->getPaypal()?->getExperienceContext()->setUserAction(ApplicationContext::USER_ACTION_CONTINUE);
+            $experienceContext = $order->getPaymentSource()?->getPaypal()?->getExperienceContext();
+
+            if ($experienceContext !== null) {
+                $experienceContext->setShippingPreference(ApplicationContext::SHIPPING_PREFERENCE_GET_FROM_FILE);
+                $experienceContext->setUserAction(ApplicationContext::USER_ACTION_CONTINUE);
+
+                // Configure shipping callback for dynamic price recalculation
+                $callbackConfig = new OrderUpdateCallbackConfig();
+                $this->router->getContext()->setScheme('https');
```

Comment:
I assume PayPal enforces `https` URLs? Otherwise, we shouldn't change the context here